components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // CreateEnvironmentVariable creates a new environment variable for the
+  // authenticated user.
+  rpc CreateEnvironmentVariable(CreateEnvironmentVariableRequest)
+      returns (CreateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserScope {
+  string repository_pattern = 1;
+}
+
+message ConfigurationScope {
+  string configuration_id = 1;
+  bool censored = 2;
+}
+
+message EnvironmentVariable {
+  string id = 1;
+  string name = 2;
+  string value = 3;
+  oneof scope {
+    UserScope user_scope = 4;
+    ConfigurationScope configuration_scope = 5;
+  }
+}
+
+message ListEnvironmentVariablesRequest {
+  string workspace_id = 1;
+  string configuration_id = 2;
+}
+
+message ListEnvironmentVariablesResponse {
+  repeated EnvironmentVariable environment_variables = 1;
+}
+
+message UpdateEnvironmentVariableRequest {
+  optional string env_var_name = 1;
```

Comment:
should an update not use the `env_var_id`?
Also do we want to allow updating the `name`?

Reply:
Also is updating the scope something we need to allow? After all users can always delete and create

Reply:
> should an update not use the env_var_id?

already fixed in latest commit